loadflow/loadflow-validation/src/main/java/com/powsybl/loadflow/validation/ShuntCompensatorsValidation.java

```diff
@@ -0,0 +1,155 @@
+            return checkShunts(network, config, writer);
+        }
+    }
+
+    public static boolean checkShunts(Network network, ValidationConfig config, Writer writer) {
+        Objects.requireNonNull(network);
+        Objects.requireNonNull(config);
+        Objects.requireNonNull(writer);
+        try (ValidationWriter shuntsWriter = ValidationUtils.createValidationWriter(network.getId(), config, writer, ValidationType.SHUNTS)) {
+            return checkShunts(network, config, shuntsWriter);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    public static boolean checkShunts(Network network, ValidationConfig config, ValidationWriter shuntsWriter) {
+        Objects.requireNonNull(network);
+        Objects.requireNonNull(config);
+        Objects.requireNonNull(shuntsWriter);
+        LOGGER.info("Checking shunt compensators of network {}", network.getId());
+        return network.getShuntStream()
+                      .sorted(Comparator.comparing(ShuntCompensator::getId))
+                      .map(shunt -> checkShunts(shunt, config, shuntsWriter))
+                      .reduce(Boolean::logicalAnd)
+                      .orElse(true);
+    }
+
+    public static boolean checkShunts(ShuntCompensator shunt, ValidationConfig config, Writer writer) {
+        Objects.requireNonNull(shunt);
+        Objects.requireNonNull(config);
+        Objects.requireNonNull(writer);
+
+        try (ValidationWriter shuntsWriter = ValidationUtils.createValidationWriter(shunt.getId(), config, writer, ValidationType.SHUNTS)) {
+            return checkShunts(shunt, config, shuntsWriter);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    public static boolean checkShunts(ShuntCompensator shunt, ValidationConfig config, ValidationWriter shuntsWriter) {
+        Objects.requireNonNull(shunt);
+        Objects.requireNonNull(config);
+        Objects.requireNonNull(shuntsWriter);
+
+        boolean validated = true;
+        float p = shunt.getTerminal().getP();
+        float q = shunt.getTerminal().getQ();
+        int currentSectionCount = shunt.getCurrentSectionCount();
+        int maximumSectionCount = shunt.getMaximumSectionCount();
+        float bPerSection = shunt.getbPerSection();
+        float qMax = shunt.getProperties().containsKey("qMax") ? Float.parseFloat((String) shunt.getProperties().get("qMax")) : Float.NaN;
+        float nominalV = shunt.getTerminal().getVoltageLevel().getNominalV();
+        Bus bus = shunt.getTerminal().getBusView().getBus();
+        if (bus != null && !Float.isNaN(bus.getV())) {
+            float v = bus.getV();
+            return checkShunts(shunt.getId(), p, q, currentSectionCount, maximumSectionCount, bPerSection, v, qMax, nominalV, config, shuntsWriter);
+        } else if (!Float.isNaN(q) && q != 0) { // if the shunt is disconnected then either “q” is not defined or “q” is 0
```

Comment:
If the shunt is disconnected, q should be NaN. 0 seems an invalid value

Reply:
The validation can work both with or without running the loadflow. It seems that the Q value of a disconnected shunt is NaN after you run the loadflow, but is 0 after the import. Should I just check that the value is NaN?

Reply:
We probably have an issue in the RTE tool which export all the IIDM files. The correct value should be NaN for disconnected injections.
Keep the test like this, we will do a fix in RTE tool and in the IIDM importer to fix these values one day